File: node/data-structures/StackTest.ts
const { Stack } = require("./Stack");

function testStack() {
    const stack = new Stack();

    console.log("Testing Stack Implementation\n---------------------------");
    // Test empty stack
    console.log("Is stack empty?", stack.isEmpty()); // true
    console.log("Pop on empty stack:", stack.pop()); // undefined

    // Push elements
    stack.push(1);
    stack.push(2);
    stack.push(3);
    console.log("Stack size after pushes:", stack.size()); // 3
    console.log("Top item (peek):", stack.peek()); // 3

    // Pop elements
    console.log("Popped item:", stack.pop()); // 3
    console.log("Stack size after pop:", stack.size()); // 2
    console.log("Is stack empty?", stack.isEmpty()); // false
    console.log("Popped item:", stack.pop()); // 2
    console.log("Popped item:", stack.pop()); // 1
    console.log("Is stack empty?", stack.isEmpty()); // true
    console.log("Pop on empty stack:", stack.pop()); // undefined
}

testStack();

File: java/data-structures/stack/StackTest.java
package stack;

public class StackTest{
  
    public static void main(String[] args) {
     Stack stack =  new Stack();
     stack.push(1);
     System.out.println(stack.pop());


     StackUsingArrayList stackUsingArrayList = new StackUsingArrayList();
     stackUsingArrayList.push(10);
     System.out.println(stackUsingArrayList.peek());
     System.out.println(stackUsingArrayList.pop());
     System.out.println(stackUsingArrayList.peek());

    }
}